### PR TITLE
feat: move executables to the main package and update docs.

### PR DESCRIPTION
**Pull request by ioga**
## Description
- reapply some work from #2108 that was reverted in #2153 and #2156.
- add more docs.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/how-to/installation/aws.txt [hunk 1]
```diff
@@ -19,11 +19,11 @@ For more information about using Determined on AWS, see the
  ``det deploy`` tool
 *********************
 
-The ``det deploy`` tool is provided by ``determined-deploy`` python
-package. It uses `AWS CloudFormation
-<https://aws.amazon.com/cloudformation/>`__ to automatically deploy and
-configure a Determined cluster. CloudFormation builds the necessary
-components for Determined into a single CloudFormation stack.
+The ``det deploy`` tool is provided by ``determined`` python package. It
```
Comment (determined-dsw) [inline docs/how-to/installation/aws.txt:22]: s/by/by the/

File: docs/how-to/installation/aws.txt [hunk 2]
```diff
@@ -51,11 +51,11 @@ can be increased by submitting a request to the `AWS Support Center
 Installation
 ============
 
-``det deploy`` command line tool can be installed using ``pip``:
```
Comment (determined-dsw) [inline docs/how-to/installation/aws.txt:54]: I'm a little confused here - det-deploy executable is moving into the determined package, but this change implies that det deploy does not exist anymore?

Comment (ioga) [reply docs/how-to/installation/aws.txt:54]: It implies the users only install `det` tool, which is what they'd use. New users shouldn't be aware `det` and `det deploy` subcommand used to be different things.

File: docs/how-to/installation/aws.txt (hunk 1 above)
Comment (ioga) [reply docs/how-to/installation/aws.txt:22]: fixed.

File: docs/how-to/installation/aws.txt (hunk 2 above)
Comment (determined-dsw) [reply docs/how-to/installation/aws.txt:54]: Fair enough.

File: docs/faq.txt [hunk 3]
```diff
@@ -75,6 +75,17 @@ version in a private registry without the right secret. Please see the
 documentation on how to configure which :ref:`version of Determined
 <configure-determined-kubernetes-version>` to install on Kubernetes.
 
+After upgrading Determined and trying to use the CLI, I get ``det: command not found`` error
+============================================================================================
+
+When upgrading from a version earlier than 0.15.0, some users may need
+to uninstall determined and then install it back again by running:
```
Comment (dzhu) [inline docs/faq.txt:82]:
```suggestion
to uninstall Determined and then install it again by running:
```

File: docs/how-to/installation/upgrades-troubleshoot.txt [hunk 4]
```diff
@@ -42,7 +42,22 @@ All users should also upgrade the CLI by running
...
+   pip install --upgrade determined
+
+.. _upgrading-from-before-0_15_0:
+
+****************************
+ Upgrading from ``<0.15.0``
+****************************
+
+When upgrading from a version earlier than 0.15.0, first uninstall all
+individual packages provided by determined, and then install the new
```
Comment (dzhu) [inline docs/how-to/installation/upgrades-troubleshoot.txt:54]:
```suggestion
individual packages provided by Determined, and then install the new
```

File: docs/release-notes/2187-executable-move.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+:orphan:
+
+**Improvements**
+
+-  **Breaking Change:** CLI: `det` and `det-deploy` executables have
```
Comment (dzhu) [inline docs/release-notes/2187-executable-move.txt:5]:
```suggestion
-  **Breaking Change:** CLI: ``det`` and ``det-deploy`` executables have
```
(Also others.)

File: docs/release-notes/2187-executable-move.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+   been moved to `determined` package, which now includes all of
```
Comment (dzhu) [inline docs/release-notes/2187-executable-move.txt:6]:
```suggestion
   been moved to the `determined` package, which now includes all
```

File: docs/release-notes/2187-executable-move.txt [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+   Determined libraries and tools. `determined-cli`,
```
Comment (dzhu) [inline docs/release-notes/2187-executable-move.txt:7]:
```suggestion
   Determined libraries and tools. The `determined-cli`,
```

File: docs/release-notes/2187-executable-move.txt [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+   `determined-deploy`, and `determined-common` packages are now
+   deprecated.
+
+   -  When upgrading from older versions, `det` command may break for
+      some users because of `pip` limitations. Please uninstall outdated
+      packages, and then reinstall determined. See more at
```
Comment (dzhu) [inline docs/release-notes/2187-executable-move.txt:13]:
```suggestion
      packages, and then reinstall Determined. See more at
```

File: docs/how-to/installation/aws.txt [hunk 9]
```diff
@@ -19,7 +19,7 @@ For more information about using Determined on AWS, see the
  ``det deploy`` tool
 *********************
 
-The ``det deploy`` tool is provided by ``determined-deploy`` python
+The ``det deploy`` tool is provided by the ``determined`` python
```
Comment (dzhu) [inline docs/how-to/installation/aws.txt:22]:
```suggestion
The ``det deploy`` tool is provided by the ``determined`` Python
```

**Review by dzhu — APPROVED** [review]
🎉

Perhaps we should rename the `harness` directory at some point.

**Pull request merged**